ast_tools/transformers/loop_unroller.py

```diff
@@ -0,0 +1,52 @@
+import ast
+from copy import deepcopy
+import astor
+from .symbol_replacer import replace_symbols
+from ..macros import unroll
+
+
+def is_call(node):
+    return isinstance(node, ast.Call)
+
+
+def is_name(node):
+    return isinstance(node, ast.Name)
+
+
+class Unroller(ast.NodeTransformer):
+    def __init__(self, env):
+        self.env = env
+
+    def flat_visit(self, body):
+        new_body = []
+        for statement in body:
+            result = self.visit(statement)
+            if isinstance(result, list):
+                new_body.extend(result)
+            else:
+                new_body.append(result)
+        return new_body
```

Comment:
This is dead code right?

Reply:
Actually, it's needed, but forgot to use it, basically, we have to recurse on the unrolled body, which might produce unrolled loops, so we use flat_visit to flatten the recursion

Reply:
Actually it might be simpler (and faster) to visit the body first, before unrolling